Meta/run.py

```diff
@@ -0,0 +1,978 @@
+            config.display_backend = "sdl,gl=off"
+        elif (
+            config.qemu_kind != QEMUKind.NativeWindows
+            and "sdl" in qemu_display_info
+            and has_virgl()
+        ):
+            config.display_backend = "sdl,gl=on"
+        elif "cocoa" in qemu_display_info:
+            config.display_backend = "cocoa,gl=off"
+        elif config.qemu_kind == QEMUKind.SerenityOS:
+            config.display_backend = "sdl,gl=off"
+        else:
+            config.display_backend = "gtk,gl=off"
+
+
+def setup_display_device(config: Configuration):
+    config.enable_gl = environ.get("SERENITY_GL") == "1"
+    provided_display_device = environ.get("SERENITY_QEMU_DISPLAY_DEVICE")
+    if provided_display_device is not None:
+        config.display_device = provided_display_device
+    elif config.enable_gl:
+        # QEMU appears to not support the GL backend for VirtIO GPU variant on macOS.
+        if config.qemu_kind == QEMUKind.MacOS:
+            raise RunError(
+                "SERENITY_GL is not supported since there's no GL backend on macOS"
+            )
+        elif config.screen_count > 1:
+            raise RunError(
+                "SERENITY_GL and multi-monitor support cannot be setup simultaneously"
+            )
+        config.display_device = "virtio-vga-gl"
+
+    elif config.screen_count > 1:
+        # QEMU appears to not support the virtio-vga VirtIO GPU variant on macOS.
+        # To ensure we can still boot on macOS with VirtIO GPU, use the virtio-gpu-pci
+        # variant, which lacks any VGA compatibility (which is not relevant for us anyway).
+        if config.qemu_kind == QEMUKind.MacOS:
+            config.display_device = f"virtio-gpu-pci,max_outputs={config.screen_count}"
+        else:
+            config.display_device = f"virtio-vga,max_outputs={config.screen_count}"
+
+        # QEMU appears to always relay absolute mouse coordinates relative to the screen that the mouse is
+        # pointed to, without any way for us to know what screen it was. So, when dealing with multiple
+        # displays force using relative coordinates only.
+        config.kernel_cmdline.append("vmmouse=off")
+
+
+def setup_boot_drive(config: Configuration):
+    provided_nvme_enable = environ.get("SERENITY_NVME_ENABLE")
+    if provided_nvme_enable is not None:
+        config.nvme_enable = provided_nvme_enable == "1"
+    provided_sd_enable = environ.get("SERENITY_USE_SDCARD")
+    if provided_sd_enable is not None:
+        config.sd_enable = provided_sd_enable == "1"
+
+    if config.architecture == Arch.Aarch64:
+        config.boot_drive = ["-drive", f"file={config.disk_image},if=sd,format=raw"]
+    elif config.nvme_enable:
+        config.boot_drive = [
+            "-drive",
+            f"file={config.disk_image},format=raw,index=0,media=disk,if=none,id=disk",
+            "-device",
+            "i82801b11-bridge,id=bridge4",
+            "-device",
+            "sdhci-pci,bus=bridge4",
```

Comment:
These would be a lot more readable if formatted so that the switch and its argument were on the same line. Other places you appended "-foo bar" to the command line directly.

Reply:
The entire file is PEP8 formatted with black, I'll have to see if it can do that in the first place. I don't find this particularly unreadable FWIW